Source/core/FileSystem.h

```diff
@@ -595,7 +597,22 @@ namespace Core {
         ~Directory();
 
     public:
-        static string Normalize(const string& input);
+        static string Normalize(const string& location)
+        {
+            // First see if we are not empy.
+            if (location.empty() == false) {
+
+                bool valid;
+                string result = File::Normalize(location, valid);
+
+                if ((valid == true) && ((result.empty() == true) || (result[result.length() - 1] != '/'))) {
+                    result += '/';
+                }
+
+                return (result);
```

Comment:
understand why you did this but it is against the coding guidelines :)
Think an easy way to correct this and not copying location into result at the beginning is just to do "string result;" at the beginning and then always return result. In case location is empty the if will not be entered and you will now return an empty result like you would have done for location. In this case it will only cost the default (empty) string constructor which should be very cheap